internal/beatcmd/beat.go

```diff
@@ -348,6 +350,11 @@ func (b *Beat) Run(ctx context.Context) error {
 		return adjustMaxProcs(ctx, 30*time.Second, logger)
 	})
 
+	slogger := slog.New(zapslog.NewHandler(logger.Core()))
+	if err := adjustMemlimit(1*time.Second, slogger); err != nil {
```

Comment:
I missed this in the first review, but isn't a 1 second refresh interval a bit on the aggressive side? I haven't delved into the details of the library, but it seems like a waste of CPU cycles.